Cabal/Distribution/Types/VersionRange.hs

```diff
@@ -130,7 +130,15 @@ withinRange v = foldVersionRange
 -- | @since 2.2
 wildcardUpperBound :: Version -> Version
 wildcardUpperBound = alterVersion $
-    \lowerBound -> init lowerBound ++ [last lowerBound + 1]
+    \lowerBound -> case unsnoc lowerBound of
+        Nothing      -> []
+        Just (xs, x) -> xs ++ [x + 1]
+
+unsnoc :: [a] -> Maybe ([a], a)
+unsnoc []     = Nothing
+unsnoc [h]    = Just ([], h)
+unsnoc (h:tl) = Just (h:mid, last1)
+    where Just (mid, last1) = unsnoc tl
```

Comment:
partial match, see my `unsnocNE` and adapt it.

Reply:
Is it a partial match?  I thought so too but I couldn't get it to fail in a REPL, but maybe I didn't try the right things?

Reply:
it doesn't fail, but it's still partial. There's no comment saying that `unsnoc tl` will always return `Just something`. You have to think, oh yes, `tl` is non-empty...